### PR TITLE
add some utility to rgb stuff to make indicators easier

### PR DESCRIPTION
**Pull request by evangs**
added ability to set multiple leds at once. 
added ability to do custom initialization of leds on power on

### CODE REVIEW

File: quantum/rgblight.h [hunk 1]
```diff
@@ -138,6 +139,7 @@ uint16_t rgblight_get_hue(void);
 uint8_t rgblight_get_sat(void);
 uint8_t rgblight_get_val(void);
 void rgblight_setrgb(uint8_t r, uint8_t g, uint8_t b);
+void rgblight_setrgb_many(LED_TYPE *ledarray, uint8_t *indexes, uint8_t led_count);
```
Comment (vomindoraan) [inline quantum/rgblight.h]: Should be `indices`; `indexes` is usually used to refer to database indexes.

**Comment by SupremeTechnopriest** [comment]
Any chance we can get this merged in?

**Comment by SupremeTechnopriest** [comment]
@drashna Okay if it works for @evangs it works for me.

**Comment by drashna** [comment]
Well, the PR that I linked isn't merged, but handles this functionality, and more. 

Specifically, the issue here, is that the rgblight_init is called later than matrix_init, and there is nothing after it that the user can hook into. 

My PR actually allows for a hook both very early in the startup process, but also very late, as well (after everything else has init'ed).  So it's not as "targeted", and will work with more features. 

That said, for either this or the other PR, you can add `.diff` to the URL and download the code changes.  Once you've done that, you can use `git apply` to apply the changes to your local repo.

**Comment by SupremeTechnopriest** [comment]
Yep Yep, sounds good.  Ill probably grab your PR today and test it out.  Thanks!

**Comment by drashna** [comment]
@SupremeTechnopriest that PR I listed is merged. 

Was this PR still useful in this case?

**Comment by skullydazed** [comment]
Does this still have value? Seems like `rgblight_setrgb_many()` might be useful if we don't already have something like that.

**Comment by drashna** [comment]
At the very least, we already have an `rgblight_set***_range` that allows for "bulk processing". 

But this allows for more stuff.   Though, given what you'd have to do, you'd set a bunch of LEDs, set them based on  the led array, and then run `rgblight_set`.   In this case, I'm not sure how useful this function really is, as I'm not really sure what you actually gain here, as opposed to doing something like: https://github.com/qmk/qmk_firmware/blob/master/users/drashna/rgb_stuff.c#L138-L161

Honestly, I think it would be better if we had better documentation for how to play with RGB Lights, over functions that .... would require very similar documentation, anyways.   

Specifically:
I don't object to this, I'm just not what the benefit of this is.

**Comment by skullydazed** [comment]
@evangs We'd like to either merge or close this one. Would you like to add some docs and maybe clean it up a little given the QMK changes since it was first submitted so we can merge this in?

**Comment by skullydazed** [comment]
I'm closing this to clean up our PR queue. We still like the idea, so if you'd like to clean it up and submit it again in the future please feel free to. :)

**Pull request closed without merging**